### PR TITLE
Added templating and access requests buttons to the new model create screen

### PR DESCRIPTION
**Pull request by ghost**
image

### CODE REVIEW

File: frontend/src/entry/CreateEntry.tsx [hunk 1]
```diff
@@ -60,6 +62,8 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
   const [collaborators, setCollaborators] = useState<CollaboratorEntry[]>(
     currentUser ? [{ entity: `${EntityKind.USER}:${currentUser?.dn}`, roles: ['owner'] }] : [],
   )
+  const [ungovernedAccess, setungovernedAccess] = useState<boolean>(false)
```
Comment (ARADDCC002) [inline frontend/src/entry/CreateEntry.tsx:65]: `boolean` and `string` can be inferred from their defaults so we don't need to declare the types, so we can just do `useState(false)`.

File: frontend/src/entry/CreateEntry.tsx [hunk 2]
```diff
@@ -174,24 +194,36 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
...
-                  data-test='publicButtonSelector'
-                />
-                <FormControlLabel
-                  value='private'
-                  control={<Radio />}
-                  label={privateLabel()}
-                  data-test='privateButtonSelector'
-                />
-              </RadioGroup>
+              <Box sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}>
```
Comment (ARADDCC002) [inline frontend/src/entry/CreateEntry.tsx:197]: We don't need to use this `<Box/>` component here

File: frontend/src/entry/CreateEntry.tsx [hunk 3]
```diff
@@ -174,24 +194,36 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
...
+                    label={publicLabel()}
+                    data-test='publicButtonSelector'
+                  />
+                  <FormControlLabel
+                    value='private'
+                    control={<Radio />}
+                    label={privateLabel()}
+                    data-test='privateButtonSelector'
+                  />
+                  <FormControlLabel
```
Comment (ARADDCC002) [inline frontend/src/entry/CreateEntry.tsx:215]: The option to allow for templating should appear at the bottom of the expansion menu under or above the choice to allow for ungoverned access. Although I can see why you'd put it under "Access control", this section is accessing the model card page - easy mistake to make though!

File: frontend/src/entry/CreateEntry.tsx [hunk 4]
```diff
@@ -224,6 +256,16 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
                     ]}
                   />
                 </Box>
+                <FormControlLabel
```
Comment (ARADDCC002) [inline frontend/src/entry/CreateEntry.tsx:259]: I would put the "allow templating" option here, and I'd suggest adding a sub-heading of "Additional settings", a bit like how the user selector appears under the heading of "Manage access list".

To easily space out different components you can use the MUI `<Stack />` component, which works like so:

```
<Stack spacing={2}>
  <Typography>This is some text!</Typography>
  <Typography>This is some more text!</Typography>
</Stack>
```

**Review by ARADDCC002 — CHANGES_REQUESTED** [review]

File: frontend/src/entry/CreateEntry.tsx [hunk 5]
```diff
@@ -130,6 +136,34 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
     )
   }
 
+  const templateLabel = () => {
```
Comment (ARADDCC002) [inline frontend/src/entry/CreateEntry.tsx:139]: `const templateLabel = useCallback(() => {`

File: frontend/src/entry/CreateEntry.tsx [hunk 6]
```diff
@@ -130,6 +136,34 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
...
+          <Typography fontWeight='bold'>Templating</Typography>
+          <Typography variant='caption'>
+            {`Allow this to be used as a template for another ${EntryKindLabel[createEntryKind]}`}
+          </Typography>
+        </Stack>
+      </Stack>
+    )
+  }
+
+  const ungovAccessLabel = () => {
```
Comment (ARADDCC002) [inline frontend/src/entry/CreateEntry.tsx:153]: `const ungovAccessLabel = useCallback(() => {`

I'd also consider renaming this to `ungovernedAccessLabel` as well

**Review by ARADDCC002 — CHANGES_REQUESTED** [review]

File: frontend/src/entry/CreateEntry.tsx [hunk 7]
```diff
@@ -224,6 +258,32 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
...
+                    control={
+                      <Switch
+                        onChange={(e) => setungovernedAccess(e.target.checked)}
+                        checked={ungovernedAccess}
+                        size='small'
+                      />
+                    }
+                  />
+                  <FormControlLabel
+                    value='false'
```
Comment (ARADDCC012) [inline frontend/src/entry/CreateEntry.tsx:276]: `value` can be removed

**Review by ARADDCC012 — CHANGES_REQUESTED** [review]

**Review by ARADDCC002 — APPROVED** [review]

File: frontend/src/entry/CreateEntry.tsx [hunk 8]
```diff
@@ -130,6 +135,42 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
     )
   }
 
+  const TemplateLabel = ({ createEntryKind }) => {
```
Comment (ARADDCC012) [inline frontend/src/entry/CreateEntry.tsx:138]: This should just be (I suggest changing the name to be clear that it in itself isn't a template):
```
const allowTemplatingLabel = useMemo(() => {
      return (
        <Stack direction='row' justifyContent='center' alignItems='center' spacing={1}>
          <FolderCopy />
          <Stack sx={{ my: 1 }}>
            <Typography fontWeight='bold'>Templating</Typography>
            <Typography variant='caption'>
              {`Allow this to be used as a template for another ${EntryKindLabel[createEntryKind]}`}
            </Typography>
          </Stack>
        </Stack>
      )
    }, [createEntryKind])
```

It's important to note that hooks in react have to be called in the same order on every render cycle, so they should always be defined at the base of the component.

I suggest having a quick read on how memo hooks work in particular, but in short this is for performance reasons. https://react.dev/reference/react/useMemo

File: frontend/src/entry/CreateEntry.tsx [hunk 9]
```diff
@@ -130,6 +135,42 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
...
+            </Typography>
+          </Stack>
+        </Stack>
+      )
+    }, [createEntryKind])
+
+    return CreateTemplateLabel
+  }
+
+  const UngovernedAccessLabel = ({ createEntryKind }) => {
```
Comment (ARADDCC012) [inline frontend/src/entry/CreateEntry.tsx:156]: Same here, this should be:
```
    const ungovernedAccessLabel = useMemo(() => {
      return (
        <Stack direction='row' justifyContent='center' alignItems='center' spacing={1}>
          <Gavel />
          <Stack sx={{ my: 1 }}>
            <Typography fontWeight='bold'>Ungoverned Access Requests</Typography>
            <Typography variant='caption'>
              {`Allow users to request access without the need for authorisation from ${EntryKindLabel[createEntryKind]} owners`}
            </Typography>
          </Stack>
        </Stack>
      )
    }, [createEntryKind])
```

File: frontend/src/entry/CreateEntry.tsx [hunk 10]
```diff
@@ -224,6 +265,31 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
...
+                  </Typography>
+                  <FormControlLabel
+                    control={
+                      <Switch
+                        onChange={(e) => setungovernedAccess(e.target.checked)}
+                        checked={ungovernedAccess}
+                        size='small'
+                      />
+                    }
+                    label={UngovernedAccessLabel({ createEntryKind })}
```
Comment (ARADDCC012) [inline frontend/src/entry/CreateEntry.tsx:280]: This should be `label={ungovernedAccessLabel}`

File: frontend/src/entry/CreateEntry.tsx [hunk 11]
```diff
@@ -224,6 +265,31 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
...
+                  />
+                  <FormControlLabel
+                    control={
+                      <Switch
+                        onChange={(event) => setAllowTemplating(event.target.checked)}
+                        checked={allowTemplating}
+                        size='small'
+                      />
+                    }
+                    label={TemplateLabel({ createEntryKind })}
```
Comment (ARADDCC012) [inline frontend/src/entry/CreateEntry.tsx:290]: This should be `label={allowTemplatingLabel}`

**Review by ARADDCC012 — CHANGES_REQUESTED** [review]

**Review by ARADDCC012 — APPROVED** [review]

**Pull request merged**